tests/components/bond/test_switch.py

```diff
@@ -0,0 +1,87 @@
+"""Tests for the Bond switch device."""
+from datetime import timedelta
+import logging
+
+from bond import DeviceTypes
+
+from homeassistant import core
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN
+from homeassistant.const import ATTR_ENTITY_ID, SERVICE_TURN_OFF, SERVICE_TURN_ON
+from homeassistant.helpers.entity_registry import EntityRegistry
+from homeassistant.util import utcnow
+
+from .common import (
+    patch_bond_device_state,
+    patch_bond_turn_off,
+    patch_bond_turn_on,
+    setup_platform,
+)
+
+from tests.common import async_fire_time_changed
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def generic_device(name: str):
+    """Create a generic device with given name."""
+    return {"name": name, "type": DeviceTypes.GENERIC_DEVICE}
+
+
+async def test_entity_registry(hass: core.HomeAssistant):
+    """Tests that the devices are registered in the entity registry."""
+    await setup_platform(hass, SWITCH_DOMAIN, generic_device("name-1"))
+
+    registry: EntityRegistry = await hass.helpers.entity_registry.async_get_registry()
+    assert [key for key in registry.entities] == ["switch.name_1"]
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5844602)